addons/website/models/ir_actions.py

```diff
@@ -2,64 +2,60 @@
 
 import urlparse
 
+from openerp import api, fields, models
 from openerp.http import request
-from openerp.osv import fields, osv
 
 
-class actions_server(osv.Model):
+class ActionsServer(models.Model):
     """ Add website option in server actions. """
     _name = 'ir.actions.server'
     _inherit = ['ir.actions.server']
 
-    def _compute_website_url(self, cr, uid, id, website_path, xml_id, context=None):
-        base_url = self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context)
-        link = website_path or xml_id or (id and '%d' % id) or ''
+    @api.one
+    def _compute_website_url(self, website_path, xml_id):
```

Comment:
It seems that the method is always invoked as: `action._compute_website_url(action.website_path, action.xml_id)`. With the new API, the parameters are useless. Maybe you could use the decorators `@api.v7` and `@api.v8` as:

    @api.v8
    def _compute_website_url(self):
        base_url = self.env['ir.config_parameter'].get_param('web.base.url')
        link = self.website_path or self.xml_id or (self.id and '%d' % self.id) or ''
        if base_url and link:
            path = '%s/%s' % ('/website/action', link)
            return '%s' % urlparse.urljoin(base_url, path)
        return ''

    @api.v7
    def _compute_website_url(self, cr, uid, id, website_path, xml_id, context=None):
        record = self.browse(cr, uid, id, context=context)
        return record._compute_website_url()

Of course this works *only* if the method is always called as above.